staging/src/k8s.io/client-go/transport/token_source_test.go

```diff
@@ -154,3 +155,83 @@ func TestCachingTokenSourceRace(t *testing.T) {
 		}
 	}
 }
+
+func TestCancelation(t *testing.T) {
+	cs := []struct {
+		name          string
+		header        http.Header
+		wrapTransport func(http.RoundTripper) http.RoundTripper
+
+		wantOauthCancel, wantBaseCancel bool
+	}{
+		{
+			name: "cancel req with bearer token skips oauth rt",
+
+			header:         map[string][]string{"Authorization": {"Bearer TOKEN"}},
+			wantBaseCancel: true,
+		},
+		{
+			name: "cancel req without bearer token hits both rts",
+
+			wantOauthCancel: true,
+			wantBaseCancel:  true,
+		},
+		{
+			name: "cancel req without bearer token hits both wrapped rts",
+			wrapTransport: func(rt http.RoundTripper) http.RoundTripper {
+				return NewUserAgentRoundTripper("testing testing", rt)
+			},
+
+			wantOauthCancel: true,
+			wantBaseCancel:  true,
+		},
+	}
+	for _, c := range cs {
+		t.Run(c.name, func(t *testing.T) {
+			baseRecorder := &recordCancelRoundTripper{}
+
+			var base http.RoundTripper = baseRecorder
+			if c.wrapTransport != nil {
+				base = c.wrapTransport(base)
+			}
+
+			oauthRecorder := &recordCancelRoundTripper{
+				base: &oauth2.Transport{
```

Comment:
You don't really need `base`. It only tests that oauth2.Transport.CancelRequest propagates to its Base. Let oauth2 tests cover that behavior.
oauth2.Transport handles https://godoc.org/golang.org/x/oauth2#Transport.CancelRequest